### PR TITLE
Support compas 2 private Data API

### PR DESCRIPTION
**Pull request by yck011522**
Supporting the new compas 2 private Data API.

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [x] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas_fab.robots.CollisionMesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by yck011522** [comment]
Note to myself that this branch is currently running against compas 2.0.0b4 with an unmerged fix:
https://github.com/compas-dev/compas/tree/bugfix/sphere_intersection

compas_robot is still under active fix

**Comment by yck011522** [comment]
At the moment, the two failing CI are:
### build / windows-ironpython
Need to change workflow to install compas_robots

### integration / ubuntu-py38-integration
doctest is failing
```
=========================== short test summary info ============================
SKIPPED [2] ../../../../../opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/doctest.py:460: all tests skipped by +SKIP option
FAILED docs/examples/03_backends_ros/03_forward_and_inverse_kinematics.rst::line:36,column:1 - roslibpy.core.RosTimeoutError: Failed to connect to ROS
FAILED docs/examples/03_backends_ros/03_forward_and_inverse_kinematics.rst::line:49,column:1 - roslibpy.core.RosTimeoutError: Failed to connect to ROS
FAILED docs/examples/03_backends_ros/03_forward_and_inverse_kinematics.rst::line:54,column:1 - roslibpy.core.RosTimeoutError: Failed to connect to ROS
FAILED docs/examples/03_backends_ros/04_plan_motion.rst::line:23,column:1 - roslibpy.core.RosTimeoutError: Failed to connect to ROS
FAILED docs/examples/03_backends_ros/04_plan_motion.rst::line:32,column:1 - roslibpy.core.RosTimeoutError: Failed to connect to ROS
FAILED docs/examples/03_backends_ros/05_collision_objects.rst::line:33,column:1 - roslibpy.core.RosTimeoutError: Failed to connect to ROS
FAILED docs/examples/03_backends_ros/05_collision_objects.rst::line:50,column:1 - roslibpy.core.RosTimeoutError: Failed to connect to ROS
FAILED docs/examples/03_backends_ros/05_collision_objects.rst::line:70,column:1 - roslibpy.core.RosTimeoutError: Failed to connect to ROS
FAILED docs/examples/05_backends_pybullet/01_pybullet_examples.rst::line:55,column:1 - AttributeError: 'Color' object has no attribute 'get_urdf_element'
```
Mostly due to geometry `__str__` no longer printing in the same way.
Except the case where the error caused by compas_robot : `AttributeError: 'Color' object has no attribute 'get_urdf_element' `

File: src/compas_fab/robots/reachability_map/reachability_map.py [hunk 1]
```diff
@@ -118,25 +119,28 @@ def points(self):
         return [f[0].point for f in self.frames]
 
     @property
-    def data(self):
-        def data_encode(obj):
+    def __data__(self):
+        def _resursive_encode(obj):
```
Comment (gonzalocasas) [inline src/compas_fab/robots/reachability_map/reachability_map.py:123]:
```suggestion
        def _recursive_encode(obj):
```

File: src/compas_fab/robots/reachability_map/reachability_map.py [hunk 2]
```diff
@@ -118,25 +119,28 @@ def points(self):
         return [f[0].point for f in self.frames]
 
     @property
-    def data(self):
-        def data_encode(obj):
+    def __data__(self):
+        def _recursive_encode(obj):
             if isinstance(obj, list):
-                return [data_encode(sub) for sub in obj]
+                return [_resursive_encode(sub) for sub in obj]
```
Comment (gonzalocasas) [inline src/compas_fab/robots/reachability_map/reachability_map.py:125]:
```suggestion
                return [_recursive_encode(sub) for sub in obj]
```

**Review by gonzalocasas — APPROVED** [review]
🚀 lgtm!

**Pull request merged**